api/scpca_portal/test/test_loader.py

```diff
@@ -0,0 +1,1450 @@
+import io
+from csv import DictReader
+from functools import partial
+from pathlib import Path
+from typing import Any, Dict, Set
+from unittest.mock import patch
+from zipfile import ZipFile
+
+from django.conf import settings
+from django.test import TransactionTestCase
+
+from scpca_portal import common, loader
+from scpca_portal.models import Project
+from scpca_portal.models.computed_file import ComputedFile
+from scpca_portal.models.library import Library
+from scpca_portal.models.sample import Sample
+
+
+class TestLoader(TransactionTestCase):
+    def setUp(self):
+        self.get_projects_metadata = partial(
+            loader.get_projects_metadata, input_bucket_name=settings.AWS_S3_INPUT_BUCKET_NAME
+        )
+        # When passing a project_id to get_projects_metadata, a list of one item is returned
+        # This lambda creates a shorthand with which to access the single returned project_metadata
+        self.get_project_metadata = lambda project_id: self.get_projects_metadata(
+            filter_on_project_id=project_id
+        )[0]
+
+        self.create_project = partial(
+            loader.create_project,
+            submitter_whitelist={"scpca"},
+            input_bucket_name=settings.AWS_S3_INPUT_BUCKET_NAME,
+            reload_existing=True,
+            update_s3=False,
+        )
+
+        self.generate_computed_files = partial(
+            loader.generate_computed_files,
+            max_workers=10,
+            update_s3=False,
+            clean_up_output_data=False,
+        )
+
+    @classmethod
+    def tearDownClass(cls):
+        super().tearDownClass()
+        # shutil.rmtree(common.OUTPUT_DATA_PATH, ignore_errors=True)
+
+    def assertObjectProperties(self, obj: Any, expected_values: Dict[str, Any]) -> None:
+        for attribute, value in expected_values.items():
+            msg = f"The actual and expected {attribute} values differ in {obj}"
+            if isinstance(value, list):
+                self.assertEqual(set(getattr(obj, attribute)), set(value), msg)
```

Comment:
use `self.assertListEqual` here. sometimes we might want to assess the sort order.